watch-library/hardware/watch/watch_private.c

```diff
@@ -245,43 +245,35 @@ void _watch_enable_usb(void) {
 
 // this function ends up getting called by printf to log stuff to the USB console.
 int _write(int file, char *ptr, int len) {
+    int written = 0;
     (void)file;
-    if (hri_usbdevice_get_CTRLA_ENABLE_bit(USB)) {
-        tud_cdc_n_write(0, (void const*)ptr, len);
-        tud_cdc_n_write_flush(0);
-        return len;
+
+    while (written < len) {
```

Comment:
Previously, if you tried to 'write' something large it would write whatever it could (tud_cdc_n_write) and then claim it had written the whole buffer.

I think it should have been enough to just properly return what was written here (i.e. the return value of tud_cdc_n_write), but in practice this failed on catting large files (just cutting off the file midway, as opposed to the earlier behaviour of cutting off each line). Suspect there's some kind of buffering in newlib that isn't behaving the way I'd expect, but doing the stupid loop here at least seems to work.